Userland/Libraries/LibWeb/CSS/Parser/Parser.cpp

```diff
@@ -4931,6 +4932,201 @@ RefPtr<StyleValue> Parser::parse_content_value(Vector<ComponentValue> const& com
     return ContentStyleValue::create(StyleValueList::create(move(content_values), StyleValueList::Separator::Space), move(alt_text));
 }
 
+// https://w3c.github.io/csswg-drafts/css-display/#the-display-properties
+RefPtr<StyleValue> Parser::parse_display_value(Vector<ComponentValue> const& component_values)
+{
+    auto is_display_outside = [](ValueID identifier) -> bool {
+        switch (identifier) {
+        case ValueID::Block:
+        case ValueID::Inline:
+        case ValueID::RunIn:
+            return true;
+        default:
+            return false;
+        }
+    };
+    auto is_display_inside = [](ValueID identifier) -> bool {
+        switch (identifier) {
+        case ValueID::Flow:
+        case ValueID::FlowRoot:
+        case ValueID::Table:
+        case ValueID::Flex:
+        case ValueID::Grid:
+        case ValueID::Ruby:
+            return true;
+        default:
+            return false;
+        }
+    };
+
+    auto parse_inside = [](ValueID identifier) -> CSS::Display::Inside {
+        switch (identifier) {
+        case ValueID::Flow:
+            return CSS::Display::Inside::Flow;
```

Comment:
The `CSS::` namespace can be removed from a lot of places in this commit. We're already inside that namespace and there isn't a danger of name conflicts. Removing it when you can makes things a bit easier to read.

(We're very inconsistent with this in places, sorry about that!)